pyproject.toml

```
@@ -1,74 +1,102 @@
 [build-system]
-requires = ["poetry-core>=1.0.0"]
-build-backend = "poetry.core.masonry.api"
+requires = ["hatchling"]
+build-backend = "hatchling.build"
 
-[tool.poetry]
+[project]
 name = "napari-wsi"
-version = "0.1.4"
-description = "A plugin to read whole slide images within napari."
-license="Apache-2.0"
+version = "1.0.0"
+description = "A plugin to read whole-slide images within napari."
+readme = "README.md"
+requires-python = ">=3.11"
+license = { file = "LICENSE" }
 authors = [
-    "Philipp Plewa <philipp.plewa@astrazeneca.com>",
+    { name = "Philipp Plewa", email = "philipp.plewa@astrazeneca.com" },
 ]
 maintainers = [
-    "Philipp Plewa <philipp.plewa@astrazeneca.com>",
-    "Robert Schweizer <robert.schweizer@astrazeneca.com>",
+    { name = "Philipp Plewa", email = "philipp.plewa@astrazeneca.com" },
+    { name = "Robert Schweizer", email = "robert.schweizer@astrazeneca.com" },
 ]
-readme = "README.md"
-repository = "https://github.com/AstraZeneca/napari-wsi"
 classifiers = [
     "Framework :: napari",
     "Intended Audience :: Developers",
     "Operating System :: OS Independent",
     "Topic :: Scientific/Engineering :: Image Processing",
 ]
+dependencies = [
+    "dask>=2025.1",
+    "magicgui>=0.10",
+    "numpy>=1.26",
+    "pillow>=11.1",
+    "universal-pathlib>=0.2",
+    "zarr>=3.0",
+]
 
-[tool.poetry.dependencies]
-python = ">=3.10,<3.12"
-dask = {version = ">=2023", extras = ["array"]}
-imagecodecs = ">=2023.9.18"
-magicgui = ">=0.7.3"
-matplotlib = ">=3.8"
-napari = ">=0.4.18"
-numpy = ">=1.26"
-rasterio = ">=1.3"
-tifffile = ">=2023"
-zarr = ">=2.16"
+[project.optional-dependencies]
+openslide = [
+    "openslide-python>=1.4",
+]
+rasterio = [
+    "rasterio>=1.4",
+]
+wsidicom = [
+    "wsidicom>=0.22",
+]
+s3 = [
+    "s3fs>=2024.12.0",
+]
+all = [
+    "napari-wsi[openslide,rasterio,wsidicom]"
```

Should this include s3?

```suggestion
    "napari-wsi[openslide,rasterio,wsidicom,s3]"
```

---

Good catch, I have removed the s3 extra. It's possible to enter an S3 URL to a DICOM image into the widget, but without having tested this feature properly, I would consider it experimental for now. Also, if one attempts this without having s3fs installed, the error message that pops up mentions the missing dependency.